### PR TITLE
[dep](thirdparty) import wasmtime library

### PR DESCRIPTION
**Pull request by taptao**
## Proposed changes

Issue Number: close https://github.com/apache/doris/pull/24504

Add wasm-lib thirdparty.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by taptao** [comment]
run buildall

**Review by zhangstar333 — DISMISSED** [review]

**Comment by taptao** [comment]
run buildall

**Comment by taptao** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by taptao** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
I encounter an issue when compiling the latest thirdparty
`cargo: command not found`
may I ask what is the typical way to install cargo conveniently?
or can you just write it in the script where you use the command cargo?

**Comment by zclllyybb** [comment]
@taptao FYI, maybe we could rethink the version of these libs because it may not comptable for some machine with old glibc version. some dependency is not avaliable.

File: thirdparty/vars.sh [hunk 1]
```diff
@@ -583,6 +595,8 @@ export TP_ARCHIVES=(
     'JSONCPP'
     'LIBUUID'
     'ALI_SDK'
+    'WASMTIME'
+    'WASMTIME_CPP'
```
Comment (adonis0147) [inline thirdparty/vars.sh:599]: Hi @taptao , it seems that these libraries don't support macOS (arm64). I think you would better remove these lines here and add them somewhere else after checking the platform. Currently, this script can not be ran successfully on macOS (arm64).

**Comment by taptao** [comment]
> I encounter an issue when compiling the latest thirdparty `cargo: command not found` may I ask what is the typical way to install cargo conveniently? or can you just write it in the script where you use the command cargo?

I installed rust through curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh

**Comment by taptao** [comment]
> @taptao FYI, maybe we could rethink the version of these libs because it may not comptable for some machine with old glibc version. some dependency is not avaliable.

yes, this is a big problem, i will rethink about it. any good suggestions?

File: thirdparty/vars.sh (hunk 1 above)
Comment (adonis0147) [reply thirdparty/vars.sh:599]: > I noticed this, but my env is mac arm, and the compilation was successful after installing rust, did you get any errors?

Hi @taptao , thanks for your prompt reply. I ran the script successfully after setting Rust up.

**Pull request merged**